tests/resources/Vsphere-UI-Util.robot

```diff
@@ -50,3 +50,108 @@ Navigate To VCH Creation Wizard
+
+    # wait until vsphere-ui service is up and running
+    Wait Until Keyword Succeeds  20x  30s  Is vSphere Client Ready  ${VCSA_IP}
+    Log To Console  vSphere Client has booted.
+    Close connection
+
+Is vSphere Client Ready
+    [Arguments]  ${VCSA_IP}
+    Log To Console  Waiting until vSphere Client is up and ready...
+    ${out}=  Run  curl -sL https://${VCSA_IP}/ui/ -k
+    ${out_len}=  Get Length  ${out}
+    Should Be True  ${out_len} > 0
+    Should Not Contain  ${out}  Service Unavailable
+    Should Not Contain  ${out}  is still initializing
+
+Cleanup Plugins From VC
+    [Arguments]  ${VC_TARGET}  ${VCSA_FINGERPRINT}  ${VC_USERNAME}=administrator@vsphere.local  ${VC_PASSWORD}=Admin!23
+    Close All Browsers
+    Log To Console  Removing VIC UI plugins from ${VC_TARGET}...
+    # remove plugins
+    ${rc}  ${uname_v}=  Run And Return Rc And Output  uname -v
+    ${is_darwin}=  Run Keyword And Return Status  Should Contain  ${uname_v}  Darwin
+    ${vic-ui-binary}=  Set Variable If  ${is_darwin}  ./vic-ui-darwin  ./vic-ui-linux
+    ${rc1}  ${out1}=  Run And Return Rc And Output  ${vic-ui-binary} remove --thumbprint ${VCSA_FINGERPRINT} --target ${VC_TARGET} --user ${VC_USERNAME} --password ${VC_PASSWORD} --key com.vmware.vic.ui
+    ${rc2}  ${out2}=  Run And Return Rc And Output  ${vic-ui-binary} remove --thumbprint ${VCSA_FINGERPRINT} --target ${VC_TARGET} --user ${VC_USERNAME} --password ${VC_PASSWORD} --key com.vmware.vic
+    Log To Console  ${out1}
+    Log To Console  ${out2}
+
+Destroy Dangling VCHs Created By Protractor
+    [Arguments]  ${VC_TARGET}  ${VCSA_FINGERPRINT}  ${VC_USERNAME}=administrator@vsphere.local  ${VC_PASSWORD}=Admin!23
+    Set Environment Variable  GOVC_URL  ${VC_TARGET}
+    Set Environment Variable  GOVC_INSECURE  1
+    Set Environment Variable  GOVC_USERNAME  ${VC_USERNAME}
+    Set Environment Variable  GOVC_PASSWORD  ${VC_PASSWORD}
+
+    ${rc}  ${uname_v}=  Run And Return Rc And Output  uname -v
+    ${is_darwin}=  Run Keyword And Return Status  Should Contain  ${uname_v}  Darwin
+    ${vic-machine-binary}=  Set Variable If  ${is_darwin}  ./ui-nightly-run-bin/vic-machine-darwin  ./ui-nightly-run-bin/vic-machine-linux
+
+    # get a list of dangling VMs created by protractor
+    ${rc}  ${out}=  Run And Return Rc And Output  govc vm.info -json true "virtual-container-host-*" | jq -r '.VirtualMachines[].Name'
+
+    # if there are any dangling VCHs, delete them. if not, exit the keyword
+    Run Keyword Unless  ${rc} == 0  Log To Console  VC inventory is clean. No need to clean up dangling VCHs
+    Run Keyword Unless  ${rc} == 0  Return From Keyword
+    @{vms}=  Split String  ${out}  \n
+    :FOR  ${vm}  IN  @{vms}
+    \  Log To Console  Destroying VCH: ${vm}
+    \  ${rc}  ${out}=  Run And Return Rc And Output  ${vic-machine-binary} delete --name ${vm} --target ${VC_TARGET} --user ${VC_USERNAME} --password '${VC_PASSWORD}' --compute-resource Cluster --force --thumbprint ${VCSA_FINGERPRINT}
```

Comment:
The only thing about this is, our wizard e2e tests should create and delete VCH's completely and by using the CLI this may be giving a false positive that they are being deleted completely even when they aren't.

Reply:
yeah, but then we are again leaving our test infrastructure vulnerable to failures because of unexpected artifacts left from previous runs. we need to confirm if the VCH has gone inside the H5 Client (maybe in the Hosts and Clusters page); as far as i know it's more challenging in Protractor to clean up our testbed than in Robot Framework.

Reply:
as long as delete failures are being reported in e2e (which should be) then that's fine.